packages/moda-sdk/src/profile/updateProfile.ts

```diff
@@ -0,0 +1,33 @@
+import { Hash } from 'viem';
+import { ProfileABI } from './abi/ProfileABI';
+import { profileAddress } from './constants/profileAddress';
+import { UpdateProfileArgs } from './types';
+
+export async function updateProfile(
+  updateProfileArgs: UpdateProfileArgs,
+): Promise<Hash> {
+  const { config, metadataUri } = updateProfileArgs;
+  const { publicClient, walletClient } = config;
```

Comment:
Lines 10-18 is used in multiple function definitions Maybe we wrap it in a helper function.

```ts
const verifyConfig = (config: Config): ConfigError | undefined => {
  const { publicClient, walletClient } = config;

   if (!publicClient) return ConfigError.PublicClientUndefined;
   if (!walletClient) return ConfigError.WalletClientUndefined;
}
```

Then in all of the functions:

```ts
const error = verifyConfig(config);
if (error) return { ok: false, error };
```